src/games/supported/Tetris.cpp

```diff
@@ -0,0 +1,133 @@
+/* *****************************************************************************
+ * ALE support for Tetris26 BIN https://github.com/udibr/tetris26
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License version 2
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ * *****************************************************************************
+ * A.L.E (Arcade Learning Environment)
+ * Copyright (c) 2009-2013 by Yavar Naddaf, Joel Veness, Marc G. Bellemare and 
+ *   the Reinforcement Learning and Artificial Intelligence Laboratory
+ * Released under the GNU General Public License; see License.txt for details. 
+ *
+ * Based on: Stella  --  "An Atari 2600 VCS Emulator"
+ * Copyright (c) 1995-2007 by Bradford W. Mott and the Stella team
+ *
+ * *****************************************************************************
+ */
+#include "Tetris.hpp"
+
+#include "../RomUtils.hpp"
+
+
+TetrisSettings::TetrisSettings() {
+
+    reset();
+}
+
+
+/* create a new instance of the rom */
+RomSettings* TetrisSettings::clone() const {
+    
+    RomSettings* rval = new TetrisSettings();
+    *rval = *this;
+    return rval;
+}
+
+
+/* process the latest information from ALE */
+void TetrisSettings::step(const System& system) {
+
+    // update the reward
+    int x = readRam(&system, 113);
+    int y = readRam(&system, 114);
+    reward_t score = 1 * (x & 0x000F) + 10 * ((x & 0x00F0) >> 4) + 100 * (y & 0x000F);
```

Comment:
Please move comments on Tetris to a sperate pull request I created on the dev branch